src/module/rules/rule-element/token-mark/rule-element.ts

```diff
@@ -0,0 +1,72 @@
+import { TokenDocumentPF2e } from "@scene/index.ts";
+import { SlugField } from "@system/schema-data-fields.ts";
+import { ErrorPF2e } from "@util";
+import { UUIDUtils } from "@util/uuid.ts";
+import type { StringField } from "types/foundry/common/data/fields.d.ts";
+import { RuleElementPF2e, RuleElementSchema, RuleElementSource } from "../index.js";
+import { MarkTargetPrompt } from "./prompt.js";
+
+/** Remember a token for later referencing */
+class TokenMarkRuleElement extends RuleElementPF2e<TokenMarkSchema> {
+    static override defineSchema(): TokenMarkSchema {
+        const { fields } = foundry.data;
+        return {
+            ...super.defineSchema(),
+            slug: new SlugField({ required: true, nullable: false, initial: undefined }),
+            uuid: new fields.StringField({ required: false, nullable: true, initial: null }),
+        };
+    }
+
+    override async preCreate({ ruleSource, itemSource, pendingItems }: RuleElementPF2e.PreCreateParams): Promise<void> {
+        if (this.ignored) return;
+
+        this.uuid &&= this.resolveInjectedProperties(this.uuid);
+
+        if (this.actor.getActiveTokens().length === 0) {
+            this.ignored = ruleSource.ignored = true;
+            return;
+        }
+
+        const token =
+            fromUuidSync(this.uuid ?? "") ??
+            (game.user.targets.size === 1
+                ? Array.from(game.user.targets)[0]!.document
+                : await new MarkTargetPrompt({ prompt: null, requirements: null }).resolveTarget());
+        if (!(token instanceof TokenDocumentPF2e)) {
+            // No token was targeted: abort creating item
+            pendingItems.splice(pendingItems.indexOf(itemSource), 1);
+            return;
+        }
+
+        this.#checkRuleSource(ruleSource);
```

Comment:
Why is this check necessary? 👀

Reply:
Because otherwise the object has no `uuid` property.

Reply:
That can cause a rule source to not have the key set? I'd assume this wouldn''t be executed if the key wasn't TokenMark already.

Reply:
It can't do that, no.